In [1]:
# Ensure your environment is activated and MySQL server is running

import pandas as pd
import pymysql
import getpass
from sqlalchemy import create_engine
from dotenv import dotenv_values


In [2]:
# Prompt for the MySQL password securely
# For testing, assign the password directly
config = dotenv_values(".env")


# SQLAlchemy connection
engine = create_engine(f"mysql+pymysql://root:{config['password']}@127.0.0.1/remote_work_db")

try:
    # Sample query to retrieve data
    query = "SELECT * FROM remote_work_mental_health LIMIT 10;"
    df = pd.read_sql(query, engine)
    display(df)
except Exception as e:
    print(f"Error: {e}")


Employee_ID  Age             Gender           Job_Role       Industry  \
0     EMP0001   32         Non-binary                 HR     Healthcare   
1     EMP0002   40             Female     Data Scientist             IT   
2     EMP0003   59         Non-binary  Software Engineer      Education   
3     EMP0004   27               Male  Software Engineer        Finance   
4     EMP0005   49               Male              Sales     Consulting   
5     EMP0006   59         Non-binary              Sales             IT   
6     EMP0007   31  Prefer not to say              Sales             IT   
7     EMP0008   42         Non-binary     Data Scientist  Manufacturing   
8     EMP0009   56  Prefer not to say     Data Scientist     Healthcare   
9     EMP0010   30             Female                 HR             IT   

   Years_of_Experience Work_Location  Hours_Worked_Per_Week  \
0                   13        Hybrid                     47   
1                    3        Remote                     52   
2                   22        Hybrid                     46   
3                   20        Onsite                     32   
4                   32        Onsite                     35   
5                   31        Hybrid                     39   
6                   24        Remote                     51   
7                    6        Onsite                     54   
8                    9        Hybrid                     24   
9                   28        Hybrid                     57   

   Number_of_Virtual_Meetings  Work_Life_Balance_Rating Stress_Level  \
0                           7                         2       Medium   
1                           4                         1       Medium   
2                          11                         5       Medium   
3                           8                         4         High   
4                          12                         2         High   
5                           3                         4         High   
6                           7                         3          Low   
7                           7                         3       Medium   
8                           4                         2         High   
9                           6                         1          Low   

  Mental_Health_Condition Access_to_Mental_Health_Resources  \
0              Depression                                No   
1                 Anxiety                                No   
2                 Anxiety                                No   
3              Depression                               Yes   
4                    None                               Yes   
5                    None                                No   
6                 Anxiety                               Yes   
7              Depression                                No   
8                    None                               Yes   
9              Depression                               Yes   

  Productivity_Change  Social_Isolation_Rating Satisfaction_with_Remote_Work  \
0            Decrease                        1                   Unsatisfied   
1            Increase                        3                     Satisfied   
2           No Change                        4                   Unsatisfied   
3            Increase                        3                   Unsatisfied   
4            Decrease                        3                   Unsatisfied   
5            Increase                        5                   Unsatisfied   
6            Decrease                        5                       Neutral   
7            Decrease                        5                     Satisfied   
8            Decrease                        2                   Unsatisfied   
9            Decrease                        2                       Neutral   

   Company_Support_for_Remote_Work Physical_Activity Sleep_Quality  \
0                                1          

1. Sample Data (First 10 Rows)

The first 10 rows of the dataset give a glimpse into the distribution of employees by attributes such as Age, Gender, Job Role, Industry, Work Location, and key indicators like Stress Level and Mental Health Condition.

Key observations:

The sample includes diverse job roles like HR, Data Scientist, Software Engineer, and Sales.
Employees are from multiple industries: IT, Healthcare, Education, Finance, etc.
Work locations vary between Hybrid, Remote, and Onsite.


In [4]:
# Query to find null values in key columns
query_2 = """
SELECT
    SUM(CASE WHEN Age IS NULL THEN 1 ELSE 0 END) AS Age_Null,
    SUM(CASE WHEN Gender IS NULL THEN 1 ELSE 0 END) AS Gender_Null,
    SUM(CASE WHEN Job_Role IS NULL THEN 1 ELSE 0 END) AS Job_Role_Null,
    SUM(CASE WHEN Industry IS NULL THEN 1 ELSE 0 END) AS Industry_Null,
    SUM(CASE WHEN Work_Location IS NULL THEN 1 ELSE 0 END) AS Work_Location_Null,
    SUM(CASE WHEN Stress_Level IS NULL THEN 1 ELSE 0 END) AS Stress_Level_Null,
    SUM(CASE WHEN Mental_Health_Condition IS NULL THEN 1 ELSE 0 END) AS Mental_Health_Null
FROM remote_work_mental_health;
"""
df_2 = pd.read_sql(query_2, engine)

# Results
display(df_2)


Age_Null  Gender_Null  Job_Role_Null  Industry_Null  Work_Location_Null  \
0       0.0          0.0            0.0            0.0                 0.0   

   Stress_Level_Null  Mental_Health_Null  
0                0.0                 0.0

2. Null Value Analysis

We checked the key columns to see if there were any missing values in Age, Gender, Job Role, Industry, Work Location, Stress Level, and Mental Health Condition.

Result:

No null values were found in the selected columns, ensuring the dataset is complete and usable for further analysis.


In [6]:
# Query to calculate the average hours worked
query_3 = "SELECT AVG(Hours_Worked_Per_Week) AS Average_Hours_Worked FROM remote_work_mental_health;"
df_3 = pd.read_sql(query_3, engine)

# The average hours worked
display(df_3)


Average_Hours_Worked
0               39.6146

3. Average Hours Worked per Week

The average number of hours worked per week across all employees is 39.6 hours.

Key Insight:

The average work week aligns closely with a standard 40-hour week, but variations in hours worked might still impact employee mental health, which we will analyze further.

In [8]:
# Query to group by work location
query_4 = """
SELECT Work_Location, COUNT(*) AS Count
FROM remote_work_mental_health
GROUP BY Work_Location;
"""
df_4 = pd.read_sql(query_4, engine)

# Display the work location count
display(df_4)


Work_Location  Count
0        Hybrid   1649
1        Remote   1714
2        Onsite   1637

4. Distribution of Employees by Work Location

The majority of employees are split between Hybrid (1,649 employees), Remote (1,714 employees), and Onsite (1,637 employees).

Key Insight:

There’s a balanced distribution across work locations, allowing for a fair comparison of mental health and productivity metrics across these groups.


In [10]:
# Query to group by work location and mental health condition
query_5 = """
SELECT Work_Location, Mental_Health_Condition, COUNT(*) AS Count
FROM remote_work_mental_health
GROUP BY Work_Location, Mental_Health_Condition;
"""
df_5 = pd.read_sql(query_5, engine)

# Results
display(df_5)


Work_Location Mental_Health_Condition  Count
0         Hybrid              Depression    421
1         Remote                 Anxiety    443
2         Hybrid                 Anxiety    428
3         Onsite              Depression    412
4         Onsite                    None    376
5         Hybrid                    None    400
6         Remote                    None    420
7         Remote              Depression    413
8         Onsite                 Burnout    442
9         Hybrid                 Burnout    400
10        Remote                 Burnout    438
11        Onsite                 Anxiety    407

5. Work Location and Mental Health Condition

We explored how work location correlates with mental health conditions such as Depression, Anxiety, Burnout, and None.

Key Results:

Hybrid workers show high numbers of depression (421), anxiety (428), and burnout (400).
Remote workers report high anxiety (443), but slightly lower depression (413).
Onsite workers show the highest numbers of burnout (442) and depression (412).
Key Insight:

Work location influences different mental health outcomes. Hybrid workers appear to struggle with a balance of both anxiety and depression, while onsite workers face high burnout levels.


In [12]:
# Query to group by stress level and work location
query_6 = """
SELECT Work_Location, Stress_Level, COUNT(*) AS Count
FROM remote_work_mental_health
GROUP BY Work_Location, Stress_Level;
"""
df_6 = pd.read_sql(query_6, engine)

# The results
display(df_6)


Work_Location Stress_Level  Count
0        Hybrid       Medium    545
1        Remote       Medium    577
2        Onsite         High    535
3        Hybrid         High    561
4        Remote          Low    547
5        Onsite       Medium    547
6        Hybrid          Low    543
7        Remote         High    590
8        Onsite          Low    555

6. Work Location and Stress Level

We grouped employees by their Stress Level across work locations.

Key Results:

Remote workers have a higher percentage reporting low stress (547) compared to onsite workers.
Hybrid workers have a mix of medium stress (545) and high stress (561).
Key Insight:

Remote work correlates more with lower stress, while hybrid workers exhibit a balanced distribution of stress levels, perhaps due to the fluctuating demands of both home and office environments.

In [14]:
# Query to group by productivity change and work location
query_7 = """
SELECT Work_Location, Productivity_Change, COUNT(*) AS Count
FROM remote_work_mental_health
GROUP BY Work_Location, Productivity_Change;
"""
df_7 = pd.read_sql(query_7, engine)

# The results
display(df_7)


Work_Location Productivity_Change  Count
0        Hybrid            Decrease    591
1        Remote            Increase    558
2        Hybrid           No Change    544
3        Onsite            Increase    514
4        Onsite            Decrease    558
5        Hybrid            Increase    514
6        Remote            Decrease    588
7        Remote           No Change    568
8        Onsite           No Change    565

7. Work Location and Productivity Change

We looked at how work location relates to Productivity Change.

Key Results:

Remote workers show more cases of increase in productivity (558) compared to onsite or hybrid workers.
Onsite workers show more cases of no change in productivity (565), while hybrid workers have more cases of a decrease in productivity (591).
Key Insight:

Remote workers seem to benefit from increased productivity, whereas hybrid workers are more prone to experience a decline in productivity.


Conclusion

The SQL analysis reinforces several key points:

Mental Health and Work Location: There’s a notable relationship between work location and mental health outcomes. 

Hybrid workers show higher levels of both anxiety and depression, while onsite workers are more prone to burnout.

Stress Levels: Remote workers report lower stress, while hybrid workers are more evenly split between medium and high stress levels.

Productivity: Remote work boosts productivity in many cases, while hybrid work may lead to decreased productivity.

This analysis provides valuable insights that align with the project’s objectives, demonstrating how work location impacts both mental health and productivity. This data can be further utilized to guide companies in improving employee well-being and work policies.

